python/guide/packages/gluon/trainer.md

```diff
@@ -1,72 +1,95 @@
-backward passes to obtain the gradients.
+The second step is the backward step which computes the gradient of the loss with respect to the parameters. In gluon, this step is  achieved by doing the first step in an `autograd.record()` scope to record the computations needed to calculate the loss, and then calling `l.backward()` to compute the gradient of the loss with respect to the parameters.
+
+The final step is to update the neural network model parameters using an optimization algorithm. In gluon this step is performed by the [`gluon.Trainer` API](http://beta.mxnet.io/api/gluon/mxnet.gluon.Trainer.html) and is the subject of this guide. The gluon trainer is instantiated with a set of learnable parameters and the optimization algorithm to learn those parameters and every training iteration, `trainer.step` updates the model parameters.
+
+## Basic Usage
+
+### Network and Trainer
+
+To illustrate how to use the gluon Trainer we will create a simple perceptron model and create a Trainer using the perceptron model parameters and a simple optimizer - 'sgd' with learning rate as 1.
 
 ```{.python .input}
-from mxnet import gluon, nd, autograd, optimizer
+from mxnet import nd, autograd, optimizer
 
-loss = gluon.loss.L2Loss()
 net = gluon.nn.Dense(1)
 net.initialize()
+
+trainer = gluon.Trainer(net.collect_params(),
+                        optimizer='sgd', optimizer_params={'learning_rate':1})
+
+```
+
+The trainer constructor also accepts the following keyword arguments for :
+
+- `kvstore` – how key value store  should be created for multi-gpu and distributed training. Check out  [`mxnet.kvstore.KVStore`](http://beta.mxnet.io/api/gluon-related/mxnet.kvstore.KVStore.html#mxnet.kvstore.KVStore) for more information. String options are any of the following ['local', 'device', 'dist_device_sync', 'dist_device_async'].
+- `compression_params` – Specifies type of gradient compression and additional arguments depending on the type of compression being used. See [mxnet.KVStore.set_gradient_compression_method](http://beta.mxnet.io/api/gluon-related/_autogen/mxnet.kvstore.KVStore.set_gradient_compression.html#mxnet.kvstore.KVStore.set_gradient_compression) for more details on gradient compression.
+- `update_on_kvstore` – Whether to perform parameter updates on kvstore. If None, then trainer will choose the more suitable option depending on the type of kvstore.
+
+### Forward and Backward Pass
+
+Before we can use the trainer we have created to update model parameters, we must first run the forward and backward passes. Here we implement a function to compute the first two steps (forward step and backward step) of training the perceptron on a random dataset.
+
+```{.python .input}
 batch_size = 8
 X = nd.random.uniform(shape=(batch_size, 4))
 y = nd.random.uniform(shape=(batch_size,))
 
+loss = gluon.loss.L2Loss()
+
 def forward_backward():
     with autograd.record():
         l = loss(net(X), y)
     l.backward()
 forward_backward()
```

Comment:
Add line space between definition and call

Reply:
done.